test/src/compose/volume-manager.spec.ts

```diff
@@ -0,0 +1,350 @@
+				async () => {
+					await expect(volumeManager.getAll()).to.eventually.deep.equal([
+						{
+							appId: 1,
+							config: {
+								driver: 'local',
+								driverOpts: {},
+								labels: {
+									'io.balena.supervised': '1',
+								},
+							},
+							name: 'redis',
+						},
+						{
+							appId: 1,
+							config: {
+								driver: 'local',
+								driverOpts: {},
+								labels: {
+									'io.balena.supervised': '1',
+								},
+							},
+							name: 'mysql',
+						},
+						{
+							appId: 1,
+							config: {
+								driver: 'local',
+								driverOpts: {},
+								labels: {},
+							},
+							name: 'backend',
+						},
+					]);
+					// Check that debug message was logged saying we found a Volume not created by us
+					expect(logDebug.lastCall.lastArg).to.equal(
+						'Found unmanaged Volume: decoy',
+					);
+				},
+				{ volumes: volumeData },
+			);
+		});
+
+		it('can parse null Volumes', async () => {
+			// Perform test with no volumes
+			await mockerode.withMockerode(async () => {
+				await expect(volumeManager.getAll()).to.eventually.deep.equal([]);
+			});
+		});
+
+		it('gets the volume for specific application', async () => {
+			// Setup volume data
+			const volumes = [
+				mockerode.createVolume({
+					Name: Volume.generateDockerName(111, 'app'),
+					Labels: {
+						'io.balena.supervised': '1',
+					},
+				}),
+				mockerode.createVolume({
+					Name: Volume.generateDockerName(222, 'otherApp'),
+					Labels: {
+						'io.balena.supervised': '1',
+					},
+				}),
+			];
+			// Perform test
+			await mockerode.withMockerode(
+				async () => {
+					await expect(
+						volumeManager.getAllByAppId(111),
+					).to.eventually.deep.equal([
+						{
+							appId: 111,
+							config: {
+								driver: 'local',
+								driverOpts: {},
+								labels: {
+									'io.balena.supervised': '1',
+								},
+							},
+							name: 'app',
+						},
+					]);
+				},
+				{ volumes },
+			);
+		});
+	});
+
+	describe('Creating volumes', () => {
+		it('creates a volume if it does not exist', async () => {
+			// Perform test
+			await mockerode.withMockerode(async (dockerode) => {
+				// Volume to create
+				const volume = Volume.fromComposeObject('main', 111, {});
+
+				// Create volume
+				await volumeManager.create(volume);
+
+				// Expect the volume to be created on the engine
+				const { Volumes } = await dockerode.listVolumes();
+				expect(Volumes).to.have.lengthOf(1);
+				expect(Volumes[0].Name).to.equal(
+					Volume.generateDockerName(111, 'main'),
+				);
+			});
+		});
+
+		it('does not try to create a volume that already exists', async () => {
+			// Setup volume data
+			const volumes = [
+				mockerode.createVolume({
+					Name: Volume.generateDockerName(111, 'main'),
+					Labels: {
+						'io.balena.supervised': '1',
+					},
+				}),
+			];
+			// Perform test
+			await mockerode.withMockerode(
+				async (dockerode) => {
+					// Volume to try again create
```

Comment:
I understand this comment's meaning when looking at the whole test suite, but it's a bit confusing by itself. Maybe something like `// Generate compose data for volume which was already set up in mock engine`

Reply:
hahaha, don't know what I was thinking when I wrote that :p